### ISSUE TITLE
关于FLUX.2-klein-base-9B+grpo训练

### ISSUE THREAD

**Issue by wanyanggu** [X-GenGroup/Flow-Factory#179]
# Environment Configuration
launcher: "accelerate"  # Options: accelerate
config_file: config/accelerate_configs/multi_gpu.yaml # Switch to config/accelerate_configs/fsdp2.yaml if you have limited GPU memory.
num_processes: 4  # Number of processes to launch (overrides config file)
main_process_port: 29500
mixed_precision: "bf16"  # Options: no, fp16, bf16

# Data Configuration
data:
  dataset_dir: "dataset/test"  # Path to dataset folder
  preprocessing_batch_size: 32  # Batch size for preprocessing
  dataloader_num_workers: 32  # Number of workers for DataLoader
  force_reprocess: false  # Force reprocessing of the dataset
  cache_dir: "dataset/test" # Cache directory for preprocessed datasets
  max_dataset_size: null  # Limit the maximum number of samples in the dataset

# Model Configuration
model:
  finetune_type: 'lora' # Options: full, lora
  lora_rank : 128
  lora_alpha : 128
  target_modules: ["attn.to_k","attn.to_q","attn.to_v","attn.to_out.0","attn.to_qkv_mlp_proj","attn.add_k_proj","attn.add_q_proj","attn.add_v_proj","attn.to_add_out","ff.linear_in","ff.linear_out","ff_context.linear_in","ff_context.linear_out","proj_mlp","proj_out"] # Options: all, default, or list of module names like ["to_k", "to_q", "to_v", "to_out.0"]
  model_name_or_path: "/apps/dat/file/llm/model/FLUX.2-klein-base-9B"  # Options: black-forest-labs/FLUX.2-klein-4B, black-forest-labs/FLUX.2-klein-9B
  model_type: "flux2-klein"
  resume_path: null # Path to load previous checkpoint/lora adapter
  resume_type: null # Options: lora, full, state. Null to auto-detect based on `finetune_type`

log:
  run_name: null  # Run name (auto: {model_type}_{finetune_type}_{trainer_type}_{timestamp})
  project: "Flow-Factory"  # Project name for logging
  logging_backend: "tensorboard"  # Options: wandb, swanlab, tensorboard, none
  save_dir: "test/"  # Directory to save model checkpoints and logs
  save_freq: 4  # Save frequency in epochs (0 to disable)
  save_model_only: true  # Save only the model weights (not optimizer, scheduler, etc.)
  
# Training Configuration
train:
  # Trainer settings
  trainer_type: 'grpo' # Options: 'grpo', 'nft', 'awm'
  advantage_aggregation: 'gdpo' # Options: 'sum', 'gdpo'
  # Clipping
  clip_range: 1.0e-4  # PPO/GRPO clipping range
  adv_clip_range: 5.0  # Advantage clipping range
  # KL div
  kl_type: 'v-based' # Options: 'x-based', 'v-based'
  kl_beta: 1.0e-3 # KL divergence coefficient. Set ~1e-2 for 'x-based' and ~1e-3 for 'v-based'.
  ref_param_device: 'cuda' # Options: cpu, cuda

  # Sampling settings
  resolution: [2048, 1536]  # Can be int or [height, width]
  condition_image_size: 1200
  num_inference_steps: 20  # Number of timesteps. Flux2-Klein has already been distilled to 4 steps.
  guidance_scale: 4.0  # Guidance scale for sampling

  # Batch and sampling
  per_device_batch_size: 1  # Batch size per device. For image-to-image task, this will always fallback to 1.
  group_size: 8  # Group size for GRPO sampling
  global_std: false  # Use global std for advantage normalization
  unique_sample_num_per_epoch: 1  # Unique samples per group
  gradient_step_per_epoch: 1  # Gradient steps per epoch
  
  # Optimization
  learning_rate: 3.0e-4 # Initial learning rate
  adam_weight_decay: 1.0e-4  # AdamW weight decay
  adam_betas: [0.9, 0.999]  # AdamW betas
  adam_epsilon: 1.0e-8  # AdamW epsilon
  max_grad_norm: 1.0  # Max gradient norm for clipping

  # EMA
  ema_decay: 0.9  # EMA decay rate (0 to disable)
  ema_update_interval: 4  # EMA update interval (in epochs)
  ema_device: "cuda"  # Device to store EMA model (options: cpu, cuda)
  
  # Gradient checkpointing
  enable_gradient_checkpointing: true  # Enable gradient checkpointing to save memory with extra compute

  # Seed
  seed: 42  # Random seed

# Scheduler Configuration
scheduler:
  dynamics_type: "ODE" 

# Evaluation settings
eval:
  resolution: [2048, 1536]  # Evaluation resolution
  condition_image_size: 1200
  per_device_batch_size: 1  # Eval batch size
  seed: 42  # Eval seed (defaults to training seed)
  guidance_scale: 4.0  # Guidance scale for sampling
  num_inference_steps: 28  # Number of eval timesteps
  eval_freq: 4  # Eval frequency in epochs (0 to disable)

# rewards:
#   - name: "pick_score"
#     reward_model: "PickScore"
#     weight: 1  # Weight of this reward model
#     batch_size: 8
#     device: "cuda"
#     dtype: bfloat16

# Optional Evaluation Reward Models
# eval_rewards:
#   - name: "text_alignment_2"
#     reward_model: "CLIP"
#     batch_size: 16
#     device: "cuda"
#     dtype: bfloat16

这是我参数配置，我发现用ODE采样出来的图会崩。知道这是什么问题吗？
